ct/listmonk.sh

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env bash
+source <(curl -s https://raw.githubusercontent.com/community-scripts/ProxmoxVE/main/misc/build.func)
+# Copyright (c) 2021-2024 community-scripts ORG
+# Author: bvdberg01
+# License: MIT
+# https://github.com/community-scripts/ProxmoxVE/raw/main/LICENSE
+
+function header_info {
+clear
+cat <<"EOF"
+    ___      __                        __  
+   / (_)____/ /_____ ___  ____  ____  / /__
+  / / / ___/ __/ __ `__ \/ __ \/ __ \/ //_/
+ / / (__  ) /_/ / / / / / /_/ / / / / ,<   
+/_/_/____/\__/_/ /_/ /_/\____/_/ /_/_/|_|  
+                                           
+EOF
+}
+header_info
+echo -e "Loading..."
+APP="listmonk"
+var_disk="4"
+var_cpu="1"
+var_ram="512"
+var_os="debian"
+var_version="12"
+variables
+color
+catch_errors
+
+function default_settings() {
+  CT_TYPE="1"
+  PW=""
+  CT_ID=$NEXTID
+  HN=$NSAPP
+  DISK_SIZE="$var_disk"
+  CORE_COUNT="$var_cpu"
+  RAM_SIZE="$var_ram"
+  BRG="vmbr0"
+  NET="dhcp"
+  GATE=""
+  APT_CACHER=""
+  APT_CACHER_IP=""
+  DISABLEIP6="no"
+  MTU=""
+  SD=""
+  NS=""
+  MAC=""
+  VLAN=""
+  SSH="no"
+  VERB="no"
+  echo_default
+}
+
+function update_script() {
+header_info
+check_container_storage
+check_container_resources
+if [[ ! -f /etc/systemd/system/listmonk.service ]]; then msg_error "No ${APP} Installation Found!"; exit; fi
+
+RELEASE=$(curl -s https://api.github.com/repos/knadh/listmonk/releases/latest | grep "tag_name" | awk '{print substr($2, 3, length($2)-4) }')
+if [[ ! -f /opt/${APP}_version.txt ]] || [[ "${RELEASE}" != "$(cat /opt/${APP}_version.txt)" ]]; then
+
+  msg_info "Stopping ${APP}"
+  systemctl stop listmonk
+  msg_ok "Stopped ${APP}"
+
+  msg_info "Updating $APP to v${RELEASE}"
+  cd /opt
+  wget -q "https://github.com/knadh/listmonk/releases/download/v${RELEASE}/listmonk_${RELEASE}_linux_amd64.tar.gz"
+  tar -xzf "listmonk_${RELEASE}_linux_amd64.tar.gz" -C /opt/listmonk
+  /opt/listmonk/listmonk --upgrade --yes --config /opt/listmonk/config.toml &>/dev/null
```

Comment:
Have you forgotten something here? Either you have to save the config.toml beforehand or it will be lost. 

The right way would be to move the config, delete the folder, download the new one, move the config back in and you're done?

Reply:
No, because the default config is not provided in the .tar.gz that is pulled from GitHub. So the config.toml will be not overwritten, but the listmonk application and also LICENSE and README.md will be overwritten.
In the listmonkl-install script you can see that the default config is generated on line 44.

Reply:
I test it after merge